### PR TITLE
[Keymap] Add pierrec83's Kyria keymap

### PR DESCRIPTION
**Pull request by pierrechevalier83**
Upstream my daily driver keymap for the Kyria.

## Description

Upstream my keymap for the kyria which uses 34 keys and uses a number of layers to provide most useful keys I use for programming and navigating my computer with minimal finger motion for optimal comfort.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kyria/keymaps/pierrec83/encoders.c [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#include QMK_KEYBOARD_H
+#include "layers.h"
+
+#ifdef ENCODER_ENABLE
+
+void encoder_update_user(uint8_t index, bool clockwise) {
+    if (index == 0) {
+        switch (biton32(layer_state)) {
```
Comment (drashna) [inline keyboards/kyria/keymaps/pierrec83/encoders.c:8]:
```suggestion
        switch (get_highest_layer(layer_state)) {
```

File: keyboards/kyria/keymaps/pierrec83/encoders.c [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+            default:
+                if (clockwise) {
+                    tap_code16(C(A(KC_RIGHT)));
+                } else {
+                    tap_code16(C(A(KC_LEFT)));
+                }
+                break;
+        }
+    } else if (index == 1) {
+        switch (biton32(layer_state)) {
```
Comment (drashna) [inline keyboards/kyria/keymaps/pierrec83/encoders.c:35]:
```suggestion
        switch (biton32(get_highest_layer)) {
```

File: keyboards/kyria/keymaps/pierrec83/readme.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+```
+mv ~/Downloads/pierrec83.json keyboards/kyria/keymaps/pierrec83/keymap.json
+```
+* Regenerate the `keymap.c`:
+```
+qmk json2c keyboards/kyria/keymaps/pierrec83/keymap.json -o keyboards/kyria/keymaps/pierrec83/keymap.c
+```
+* Flash the firmware (for instance, if left hand is plugged):
+```
+sudo qmk flash -kb kyria -km pierrec83 -bl dfu-split-left
```
Comment (drashna) [inline keyboards/kyria/keymaps/pierrec83/readme.md:39]: shouldn't need/use sudo.

**Comment by pierrechevalier83** [comment]
Thanks for the review, @drashna. I have addressed your comments :smiley:

**Pull request merged**